[#147] feat: Modify the project metadata

IanSeng opened the pull request:
# Summary 
This pr is to implement the feature to allow user to modify the project metadata. The metadata will be updated according to the input from the user.

**Use cases/Implementation Logic**:
- To avoid overstraining the system, the metadata will only be changed when the input is not being focused.
- If a language does not exist in Keyman database, it will still allow user to input the langue name. 
- A dispatch function is used to allow children components to communicate with parents component for a better reusability of input component as well as code duplication. 

**Note**: Since this other PRs have dependency on this pr, the pr review order should ideally in the sequence of [This PR](https://github.com/eddieantonio/predictive-text-studio/pull/169)> [#168](https://github.com/eddieantonio/predictive-text-studio/pull/177) > [#165](https://github.com/eddieantonio/predictive-text-studio/pull/178). Feel free to ping me if there is any questions ☺️
## Sub-tasks 
 - [x] Language name — Important: let the use choose the name of their language, regardless of what already exists in the Keyman database!
 - [x] Author name/organization 
 - [x] copyright string
 - [x] BCP-47 tag (maybe make this hidden or an advanced option?)

## Issue 
Resolves #147

IanSeng committed:
even on change event

IanSeng committed:
implementation:

IanSeng committed:
emit function value

IanSeng committed:
implement indexdDB update in autocomplete

IanSeng committed:
remove input placeholder

IanSeng committed:
code clean up

IanSeng committed:
code clean up

IanSeng committed:
able to unkown langauge

eddieantonio commented on src/app/components/LanguageInfo.svelte:
```diff
@@ -1,7 +1,15 @@
+  // Any is used here to store temporary dynamic object
+  let tempObj: any = {};
+
+  function onInputValue(event: CustomEvent) {
```
Perhaps rename this to `updateMetadata`?

```suggestion
  function updateMetadata(event: CustomEvent) {
```

That way, it's clear what is happening in `<AutoComplete>` and `<InputField>`:

`<InputField on:message={updateMetadata} id="authorName">`

eddieantonio reviewed (approved):
LGTM! Nice use of custom event dispatchers!

IanSeng committed:
address feedback

IanSeng commented:
> LGTM! Nice use of custom event dispatchers!
My apology for not being able to address the feedback immediately due to having other commitments. I have addressed the feedback accordingly and #168 is ready to be review in a couple of mins. 🎉
